### PR TITLE
HDDS-7220. SCM should use sub-ca certificate for token signature without HA enabled

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/HDDS-7220

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 1]
```diff
@@ -788,45 +788,45 @@ private void initializeCAnSecurityProtocol(OzoneConfiguration conf,
...
         rootCertificateServer = null;
       }
     } else {
-      // On a upgraded cluster primary scm nodeId will not be set as init will
-      // not be run again after upgrade. So for a upgraded cluster where init
-      // has not happened again we will have setup like before where it has
+      // On an upgraded cluster primary scm nodeId will not be set as init will
+      // not be run again after upgrade. So for an upgraded cluster where init
+      // has not happened again we will have to set up like before where it has
       // one CA server which is issuing certificates to DN and OM.
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:826]: This comment no longer seems to be true, as now will have two servers, root-CA and sub-CA.

**Review by adoroszlai — COMMENTED** [review]
Thanks @ChenSammi for working on this.  Have you tested upgrade in secure non-HA environment?  (AFAIK we do not have acceptance test for upgrade in secure cluster.)

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java (hunk 1 above)
Comment (ChenSammi) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:826]: Thanks @adoroszlai , let me see if we can add a acceptance test for upgrade in secure cluster.

Comment (ChenSammi) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:826]: Test the non-HA upgrade locally with single node cluster.

File: hadoop-ozone/dist/src/main/compose/ozonesecure/docker-config [hunk 2]
```diff
@@ -35,6 +35,7 @@ OZONE-SITE.XML_hdds.grpc.tls.enabled=true
 OZONE-SITE.XML_ozone.replication=3
 OZONE-SITE.XML_ozone.datanode.pipeline.limit=1
 OZONE-SITE.XML_hdds.scmclient.max.retry.timeout=30s
+OZONE-SITE.XML_ozone.scm.ratis.enable=false
```
Comment (fapifta) [inline hadoop-ozone/dist/src/main/compose/ozonesecure/docker-config:38]: Unsure if we really need these here, what is the reason to disable ratis for this docker env and for the non-HA one?

**Review by fapifta — COMMENTED** [review]
I am +1 for making this change, thank you for tracking this down @ChenSammi.

Please check my comment on the ratis enablement flag for the integration tests, if that is necessary then please do not hesitate to commit, if that has just remained there please remove it, and we are good to go from my point of view.

File: hadoop-ozone/dist/src/main/compose/ozonesecure/docker-config (hunk 2 above)
Comment (ChenSammi) [reply hadoop-ozone/dist/src/main/compose/ozonesecure/docker-config:38]: Right, this ozonesecure directory is supposed for the non-HA case.  There is another ozonesecure-ha directory for the HA case.

**Review by fapifta — APPROVED** [review]

**Review by fapifta — CHANGES_REQUESTED** [review]
As we discussed, to turn off ratis is necessary to verify the patch, but in general we have ratis enabled by default, and so that we should not turn it off for the future as we discussed with @ChenSammi 

So let's commit this without disabling ratis, and later on we might think about how to implement testing for both ratis enabled and disabled case in non-HA environments.

**Review by fapifta — APPROVED** [review]
+1 commiting it once the CI is green. Thank you Sammi to take a look into the config for tests ;)

**Comment by ChenSammi** [comment]
Thanks @fapifta for the code review.  I will merge it since CI is green now.

**Pull request merged**